src/mscorlib/src/System/BitConverter.cs

```diff
@@ -457,7 +457,31 @@ unsafe public static double ToDouble (byte[] value, int startIndex)
             // If we ever run on big endian machines, produce two versions where endianness is specified.
             Contract.Assert(IsLittleEndian, "This method is implemented assuming little endian with an ambiguous spec.");
             return *((double*)&value);
-        }                    
+        }
+
+        [SecuritySafeCritical]
+        public static unsafe int SingleToInt32Bits(float value)
+        {
+            // If we're on a big endian machine, what should this method do?  You could argue for
+            // either big endian or little endian, depending on whether you are writing to a file that
+            // should be used by other programs on that processor, or for compatibility across multiple
+            // formats.  Because this is ambiguous, we're excluding this from the Portable Library & Win8 Profile.
+            // If we ever run on big endian machines, produce two versions where endianness is specified.
+            Contract.Assert(IsLittleEndian, "This method is implemented assuming little endian with an ambiguous spec.");
+            return *((int*)&value);
+        }
+
+        [SecuritySafeCritical]
+        public static unsafe float Int32BitsToSingle(int value)
+        {
+            // If we're on a big endian machine, what should this method do?  You could argue for
```

Comment:
If I was writing it from scratch: I wouldn't have included it. I included
it because it was exactly as relevant or irrelevant as the corresponding
comment in the 64-bit method. I was going for consistency.
On 27 Apr 2015 2:44 pm, "Sam Harwell" <notifications@github.com> wrote:

> In src/mscorlib/src/System/BitConverter.cs
> <https://github.com/dotnet/coreclr/pull/833#discussion_r29146605>:
>
> > +        [SecuritySafeCritical]
> > +        public static unsafe int SingleToInt32Bits(float value)
> > +        {
> > +            // If we're on a big endian machine, what should this method do?  You could argue for
> > +            // either big endian or little endian, depending on whether you are writing to a file that
> > +            // should be used by other programs on that processor, or for compatibility across multiple
> > +            // formats.  Because this is ambiguous, we're excluding this from the Portable Library & Win8 Profile.
> > +            // If we ever run on big endian machines, produce two versions where endianness is specified.
> > +            Contract.Assert(IsLittleEndian, "This method is implemented assuming little endian with an ambiguous spec.");
> > +            return *((int*)&value);
> > +        }
> > +
> > +        [SecuritySafeCritical]
> > +        public static unsafe float Int32BitsToSingle(int value)
> > +        {
> > +            // If we're on a big endian machine, what should this method do?  You could argue for
>
> [image: :question:] Why include this comment? It's unlikely that the .NET
> Framework will ever support a system which uses different endianness for
> integers and floating point values (no one makes such a system anymore). In
> all other cases, this method behave as expected on both little- and
> big-endian machines.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dotnet/coreclr/pull/833/files#r29146605>.
>